### ISSUE TITLE
Alignment results vary depending on read names?

### ISSUE THREAD

**Issue by mlovci** [BenLangmead/bowtie#64]
I am finding that varying the read name of a .fasta entry causes differences in the output number of multi-mapping sites reported by `bowtie`.

Here's an example input:

```
>p
TCTTGGATACGATATATAAGAAAAATG
>p1
TCTTGGATACGATATATAAGAAAAATG
>p2
TCTTGGATACGATATATAAGAAAAATG
>p12
TCTTGGATACGATATATAAGAAAAATG
```

I aligned to an index built with bowtie-build version 1.1.0 and the alignment was tested with bowtie 1.1.0 and 1.2.1.1.

The sequence for the genome we used is [here](https://phytozome.jgi.doe.gov/pz/portal.html#%21bulk?org=Org_Ghirsutum_er).

Here is the output of the number of reported reads for each name:

```
$ bowtie --verbose -k 10 -f -S -p 10 $INDEXPATH/Ghirsutum_458_v1.0 tmp.fa |grep -v "@" |cut -f 1 |sort |uniq -c
INFO: Command: /usr/bin/bowtie/bowtie-align-s --wrapper basic-0 -k 10 -f -S -p 10 $INDEXPATH/Ghirsutum_458_v1.0 tmp.fa
# reads processed: 4
# reads with at least one reported alignment: 4 (100.00%)
# reads that failed to align: 0 (0.00%)
Reported 21 alignments to 1 output stream(s)
      3 p
      9 p1
      6 p12
      3 p2
```

Is this the expected behavior?

**Comment by BenLangmead** [comment]
Because the pseudo-random generator is being seeded differently for each of the four reads, it's generally not surprising when there are differences in the alignments that result.  In your case, you are seeing different _numbers_ of alignments for each, which does surprise me a little.  It may simply be due to the pseudo-random generator, but I think we should check.  Is there any way you can share that index with us?

**Comment by BenLangmead** [comment]
Never mind -- I was able to get the FASTA from the JGI site.  We will look

**Comment by mlovci** [comment]
Thanks for looking into it!

**Comment by anvlasova** [comment]
hi! Do you have any update on this? 

I found similar problem; but in my case the fact of pair being aligned or not depend on the read name...

**Comment by ch4rr0** [comment]
As Ben touched upon, this issue was indeed related to the pseudo-random number generator being seeded differently for the four reads. The seed for each read gets calculated using the read name, quality value, sequence and global seed value. This would result in identical reads getting assigned different values. In my testing, I changed the behavior so that only the sequence, quality value, and global seed would be considered. With this change in place i got the following output:

```
# reads processed: 4
# reads with at least one alignment: 4 (100.00%)
# reads that failed to align: 0 (0.00%)
Reported 24 alignments
   6 p0
   6 p1
   6 p2
   6 p3
````

**Comment by ch4rr0** [comment]
This change was released as part of v1.3.0.
